tap_mambu/helpers/client.py

```diff
@@ -80,33 +102,28 @@ class MambuNoAuditApikeyInConfig(MambuError):
     500: MambuInternalServiceError}
 
 
-def get_exception_for_error_code(error_code):
-    return ERROR_CODE_EXCEPTION_MAPPING.get(error_code, MambuError)
-
-
 def raise_for_error(response):
-    LOGGER.error('ERROR {}: {}, REASON: {}'.format(response.status_code,
-                                                   response.text, response.reason))
+    """
+    Raises the associated response exception.
+    :param resp: requests.Response object
+    """
     try:
         response.raise_for_status()
     except (requests.HTTPError, requests.ConnectionError) as error:
         try:
-            content_length = len(response.content)
-            if content_length == 0:
-                # There is nothing we can do here since Mambu has neither sent
-                # us a 2xx response nor a response content.
-                return
-            response = response.json()
-            if ('error' in response) or ('errorCode' in response):
-                message = '%s: %s' % (response.get('error', str(error)),
-                                      response.get('message', 'Unknown Error'))
-                error_code = response.get('status')
-                ex = get_exception_for_error_code(error_code)
-                raise ex(message)
-            else:
-                raise MambuError(error)
-        except (ValueError, TypeError):
-            raise MambuError(error)
+            error_code = 500 if response.status_code > 500 else response.status_code
```

Comment:
Are we treating all 500+ error codes as 500? That doesn't look like a correct approach.

Reply:
Logically we are not altering the request based on any http status code (apart from retrying any exception that we come across) this will only be helpful for generating appropriate log statements, but i don't find it necessary to add more error codes right now, as mambu has over 10,000 error codes, https://support.mambu.com/docs/api-response-error-codes, and those are not the http standard error codes.